ui/src/components/nfa-icon/nfa-icon.tsx

```diff
@@ -0,0 +1,17 @@
+import { parseNumberToHexColor } from '@/utils/color';
+import { NFAIconStyles as NS } from './nfa-icon.styles';
+
+type NFAIconProps = {
+  image: string;
+  color: number;
+};
+
+export const NFAIcon: React.FC<NFAIconProps> = ({ image, color }) => {

```

Comment:
This is a not customizable component, maybe it should be a fragment in the view folder.